test/unit/host_test.rb

```diff
@@ -686,6 +686,32 @@ def teardown
       assert !host.valid?
     end
 
+    test 'should succeed validation if owner not set' do
+      host = Host.new :name => "myfullhost", :mac => "aabbecddeeff", :ip => "3.3.4.03", :medium => media(:one),
+        :domain => domains(:mydomain), :operatingsystem => operatingsystems(:redhat), :subnet => subnets(:two), :puppet_proxy => smart_proxies(:puppetmaster),
+        :architecture => architectures(:x86_64), :environment => environments(:production), :managed => true
+      assert host.valid?
+    end
+
+    test "should not save if owner_type is set without owner" do
+      host = Host.new :name => "myfullhost", :mac => "aabbecddeeff", :ip => "3.3.4.03", :medium => media(:one),
+        :domain => domains(:mydomain), :operatingsystem => operatingsystems(:redhat), :subnet => subnets(:two), :puppet_proxy => smart_proxies(:puppetmaster),
+        :architecture => architectures(:x86_64), :environment => environments(:production), :managed => true,
+        :owner_type => "Usergroup"
+      assert !host.valid?
```

Comment:
why not use `refute_valid` and `assert_valid` in these tests? or at least `refute host.valid?`